linkerd/app/outbound/src/server.rs

```diff
@@ -55,19 +55,62 @@ where
     P::Error: Send,
 {
     let tcp_balance = tcp::balance::stack(&config.proxy, tcp_connect.clone(), resolve);
-    stack_with_tcp_balancer(
+    let accept = accept_with_tcp_balancer(
         config,
         profiles,
         tcp_connect,
         tcp_balance,
         http_router,
-        metrics,
+        metrics.clone(),
         span_sink,
         drain,
-    )
+    );
+    cache_accept(&config.proxy, metrics, accept)
 }
 
-pub fn stack_with_tcp_balancer<P, C, T, H, S, I>(
+pub fn cache_accept<N, S, I>(
```

Comment:
Open to bikeshed the name of this function (and `accept_with_tcp_balancer`), I had to split these out to sneak in the tracking layer at the right point in the stack.

Reply:
How about just `cache` for this? There's nothing to disambiguate this with, given that it's in the `server` module. And for `accept_with_tcp_balancer`, how about just `stack`?